service/src/integrationTest/kotlin/fi/espoo/evaka/serviceneed/ServiceNeedQueriesTest.kt

```diff
@@ -14,4 +18,20 @@ class ServiceNeedQueriesTest : PureJdbiTest(resetDbBeforeEach = true) {
     fun getServiceNeedOptionPublicInfos() {
         assertThat(db.read { tx -> tx.getServiceNeedOptionPublicInfos(PlacementType.values().toList()) }).isEmpty()
     }
+
+    @Test
+    fun getOnlyShownServiceNeedOptionPublicInfos() {
+        db.transaction { tx -> tx.insertServiceNeedOptions() }
+        db.transaction { tx -> tx.updateShowForCitizen() }
+        val queriedOptions = db.read { tx -> tx.getServiceNeedOptionPublicInfos(PlacementType.values().toList()) }
+        // I couln't get any of the fancy asssertj list assertions to work in Kotlin, so test the stupid way
+        queriedOptions.forEach {
+            assertThat(it.id).isNotEqualTo(snPreschoolDaycare45.id)
+            assertThat(it.id).isNotEqualTo(snPreparatoryDaycare50.id)
+        }
+    }
+
+    fun Database.Transaction.updateShowForCitizen() {
+        createUpdate("UPDATE service_need_option SET show_for_citizen=FALSE WHERE id='" + snPreschoolDaycare45.id + "' OR id='" + snPreparatoryDaycare50.id + "'").execute()
```

Comment:
I know there is no risk of sql injection here, but still using a parameterized query is a good idea for consistency. You can use eg. 
```
execute(UPDATE service_need_option SET show_for_citizen=FALSE WHERE id=? OR id=?", snPreschoolDaycare45.id, snPreparatoryDaycare50.id)
```

Reply:
OK